src/data/rat/basic.lean

```diff
@@ -562,6 +573,30 @@ theorem mk_pnat_denom (n : ℤ) (d : ℕ+) :
   (mk_pnat n d).denom = d / nat.gcd n.nat_abs d :=
 by cases d; refl
 
+lemma num_mk (n d : ℤ) :
+  (n /. d).num = if d = 0 then 0 else d.sign * n / n.gcd d :=
+begin
+  cases d,
+  { cases d,
```

Comment:
This `d` is autogenerated from the `cases` in the line above, could you name the variable there explicitly?